test/controllers/thesis_controller_test.rb

```diff
@@ -353,4 +353,35 @@ class ThesisControllerTest < ActionDispatch::IntegrationTest
     assert_equal path, thesis_process_path(theses(:one))
   end
 
+  test 'removing a file from a thesis will be reflected with a link to its transfer in the flash message' do
+    sign_in users(:processor)
+    f = Rails.root.join('test','fixtures','files','a_pdf.pdf')
+    tr = transfers(:valid)
+    tr.files.attach(io: File.open(f), filename: 'a_pdf.pdf')
+    tr.save
+    tr.reload
+    th = theses(:publication_review_except_hold)
+    th.files.attach(tr.files.first.blob)
+    th.save
+    th.reload
+    blob_count = ActiveStorage::Blob.count
+    patch thesis_process_update_path(theses(:publication_review_except_hold)),
+      params: {
+        thesis: {
+          "title": "My Almost-Ready Thesis",
+          "issues_found": "true",
+          "files_attachments_attributes": {
+            "0": {
+                "id": th.files.first.id,
+                "_destroy": 1
+              }
+          }
+        }
+      }
+    follow_redirect!
+    # The flash message has a link back to the transfer with the file.
+    assert_select 'div.alert ul li a', text: 'a_pdf.pdf', href: transfer_path(th), count: 1
```

Comment:
It's a little unclear what is happening in this test. I feel like it attaches then removes a single file, but this assertion is checking for the presence of the file name which feels confusing (I think it's checking for it in the flash, but if it was still attached I think this would still pass due to it showing up in the record itself?